README.md

```diff
@@ -1,29 +1,555 @@
+}
+```
+
+`EcdsaKeypairFactory` and `Ed25519KeypairFactory` also conform to `DerivableSeedFactoryProtocol` so you can derive child seed from a parent seed
+```swift
+public protocol DerivableSeedFactoryProtocol: KeypairFactoryProtocol {
+    func deriveChildSeedFromParent(_ seed: Data,
+                                   chaincodeList: [Chaincode]) throws -> Data
+}
+```
+
+All components usage example:
+```swift
+let junctionFactory = SubstrateJunctionFactory()
+let seedFactory = SeedFactory()
+let keypairFactory = SR25519KeypairFactory()
+
+let path = "//foo/bar"
+let mnemonic = "bottom drive obey lake curtain smoke basket hold race lonely fit walk"
+
+let junctionResult = try junctionFactory.parse(path: path)
+
+let seedResult = try seedFactory.deriveSeed(
+    from: mnemonic,
+    password: junctionResult.password ?? "")
+
+let keypair = try keypairFactory.createKeypairFromSeed(
+    seedResult.seed.miniSeed,
+    chaincodeList: junctionResult.chaincodes
+)
+
+let publicKey = keypair.publicKey().rawData()
+let privateKey = keypair.privateKey().rawData()
+```
+
+### Keystore
+
+
+### Icon
+One of the features of Polkadot UI is a special icon, generated from public key data. Currently, there is [TypeScript implementation](https://github.com/polkadot-js/ui/blob/master/packages/ui-shared/src/icons/polkadot.ts) and we just re-implemented it for iOS. Generally, it just packs smaller circles inside the hexagon inscribed into an outer circle with a radius of 32 points. Colors to fill small circles are chosen based on binary representation of the public key.
+
+Usage:
+```swift
+let iconGenerator = PolkadotIconGenerator()
+let address = "5Dqvi1p4C7EhPPFKCixpF3QiaJEaDwWrR9gfWR5eUsfC39TX"
+let icon = try? iconGenerator.generateFromAddress(address)
+```
+
+Icons example:
+
+<img src="https://user-images.githubusercontent.com/3176149/132311408-c8e02e33-eb51-4f35-bdde-76b72ba07333.png" height="50">
+
+### QR
+QR is one of the most common ways to pass wallet addresses and public keys from person to person. To create or read QR-codes you will need to create an instance of SubstrateQREncoder or SubstrateQRDecoder and call corresponding functions.
+
+#### Encoding
+```swift
+let substrateEncoder = SubstrateQREncoder()
+
+let info = SubstrateQRInfo
+(
+    address: address,
+    rawPublicKey: publicKey,
+    username: username
+)
+
+let result = try? substrateEncoder.encode(info: info)
+```
+
+#### Decoding
+```swift
+let substrateDecoder = SubstrateQRDecoder(chainType: 2) // Kusama network
+
+let info = try? substrateDecoder.decode(data: data)
+let address = info.address
+```
+
+#### Additional info
+Following is the format of a QR that is used for transfers:
+
+`<prefix>:<address>:<public_key in hex>:<name>`, where **prefix** = "substrate"
+
+Example:
+
+`substrate:FiLhWLARS32oxm4s64gmEMSppAdugsvaAx1pCjweTLGn5Rf:0x8ad2a3fba73321961cd5d1b8272aa95a21e75dd5b098fb36ed996961ac7b2931:Russel`
+
+
+### Common
```

Comment:
I think Common is redundant here because it is private part. We could just add swift docs in corresponding files.

Reply:
I support your suggestion in general, but some parts are public. I suggest to leave it here for now, cleaning out redundant parts and remove it after we start doing swift docs. What do you think?